plugins/destination/mongodb/client/migrate.go

```diff
@@ -2,11 +2,114 @@ package client
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/specs"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
-// MongoDB. No migrations needed :)
-func (*Client) Migrate(context.Context, schema.Tables) error {
+func (c *Client) Migrate(ctx context.Context, tables schema.Tables) error {
+	for _, t := range tables {
+		if err := c.migrateTable(ctx, t); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *Client) migrateTable(ctx context.Context, table *schema.Table) error {
+	indexModels := c.getIndexTemplates(table)
+	for _, mdl := range indexModels {
+		res, err := c.client.Database(c.pluginSpec.Database).Collection(table.Name).Indexes().CreateOne(ctx, mdl)
```

Comment:
There's `CreateMany` as well but it would get complicated to do the error handling afterwards. It fails on the first error and you'd have to do re-runs to check/handle all.